src/commands/app/upgrade.tsx

```diff
@@ -227,3 +274,61 @@ async function forceTargetVersionSelection(
     targetAppVersionString,
   );
 }
+
+async function updateMissingSystemSoftwareDependency(
+  process: ProcessRenderer,
+  apiClient: MittwaldAPIV2Client,
+  appInstallationId: string,
+  dependency: AppSystemSoftwareDependency,
+  force: boolean,
+) {
+  const dependencySoftware = await apiClient.app.getSystemsoftware({
+    systemSoftwareId: dependency.systemSoftwareId,
+  });
+  assertSuccess(dependencySoftware);
+
+  const dependencyVersionList = await apiClient.app.listSystemsoftwareversions({
+    systemSoftwareId: dependency.systemSoftwareId,
+  });
+  assertSuccess(dependencyVersionList);
+
+  let dependencyTargetVersion: AppSystemSoftwareVersion = {
+    id: "not yet set",
+    externalVersion: "0.0.0",
+    internalVersion: "0.0.0",
+  };
+
+  for (const dependencyVersion of dependencyVersionList.data) {
+    if (
+      semver.satisfies(
+        dependencyVersion.internalVersion,
+        dependency.versionRange,
+      ) &&
+      semver.gt(
+        dependencyVersion.internalVersion,
+        dependencyTargetVersion.internalVersion,
+      )
+    ) {
+      dependencyTargetVersion = dependencyVersion;
+    }
+  }
+
+  if (!force) {
+    const confirmed: boolean = await process.addConfirmation(
```

Comment:
Possible UX optimization: Maybe only prompt for confirmation _once_, for all proposed system software updates, and not once for each?